### DISCUSSION TITLE
Using #[Url] attribute with history option set to true doesn't update url unless calling $wire.$refresh()

### DISCUSSION THREAD

**Discussion by rooselle** [livewire/livewire#7583] [7. Bugs (archived)]
### Livewire version

3.3.3

### Laravel version

10.38.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create (or update) a Livewire component with a property with attribute `#[Url(history: true)]`
2. On the related template, create a button which, on click, updates the previously created property
3. The values of the property is well changed, but nothing shows in the url
4. Add `$wire.$refresh()` and click on the button again: the property value is displayed in the url

Example component:
```php
<?php

namespace App\Livewire;

use Livewire\Attributes\Url;
use Livewire\Component;

class Test extends Component
{
    #[Url(history: true)]
    public string $param = '';

    public function render()
    {
        return view('livewire.test');
    }
}
```

Example template:
```php
<div>
    <div x-data="foo">
        <button x-on:click="changeParam">Change param</button>
    </div>
</div>

@script
<script type="text/javascript">
    Alpine.data('foo', () => {
        return {
            changeParam() {
                console.log($wire.param);
                $wire.param = (Math.random() + 1).toString(36).substring(7);
                // $wire.$refresh();
                console.log($wire.param);
            },
        }
    });
</script>
@endscript
```

### Repository or Wirebox URL

this doesn't work on Wirebox

### Additional information

I've opened a new discussion as a bug because I don't know if it's actually a bug, or if this behaviour is wanted and I should rather submit on issue on livewire/docs (to add the `$wire.refresh()` bit to the doc).

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by troccoli** [comment]
I had a similar issue.

I have build a CRUD page, when in the index page I have a button to add a new item.

The data can be filtered and the filter works fine: when I change the filter the data is refreshed and the URL changes. However, the URL is not persisted in the browser's history. This means that when I try and add a new item and then press the back button (or simulate it with `history.back()` I am sent back to the original page, without the filter applied.

I do have a workaround, FWIW. 

To update the data when the filter changes I use events:
- the user chooses a filter (it's a dropdown)
- an event to dispatched
- the Index component is listening to the event and updates the data

The above was the setup that did not work. What I did to make it works is the following:
- the user chooses a filter
- an event is dispatched TO THE FILTER (I use `$dispatchSelf()`)
- the filter component is listening to that event and it simply dispatches the original event, nothing else
- the Index component is listening to the original event and updates the data

My guess is that because the Filter component is involved then it's refreshed and the URL is persisted in the history.

**Comment by lrljoe** [comment]
use the set method to tell Livewire to send it to the backend

**Reply by troccoli** [reply]
@lrljoe could you expand on that?

**Reply by danie-ramdhani** [reply]
```js
$wire.$set('param', (Math.random() + 1).toString(36).substring(7)); // live update
```

https://livewire.laravel.com/docs/javascript#the-wire-object

**Reply by troccoli** [reply]
Ah I see.

This may well work for @rooselle's case, but not for mine.

**Reply by rooselle** [reply]
Thanks @lrljoe & @danie-ramdhani ! It's been so many months that I actually don't remember why I needed that. :joy:  But I'll keep it in mind if I come across this in the future.
